### DISCUSSION TITLE
Support and tricks thread for 0.8.0-RC1

### DISCUSSION THREAD

**Discussion by tubalainen** [blakeblackshear/frigate#536] [General]
Hi!

179 commits from 0.7.3 - lots of discussions and NOW we have RC1! Waaahooo!!!!!

https://github.com/blakeblackshear/frigate/tree/release-0.8.0

Just migrated to the new RC1 release and it works GREAT so far!
The new UI is fantastic! A superspecial thanks for all the hard work @blakeblackshear @paularmstrong you guys ROCK!

Will try to help if I can if anyone runs into issues with the migration.

Some tips:
1.) Update the HASS Addon after upgrading Frigate
2.) The "mask" part, dont forget to remove "poly" from your line (only polys are supported from now)
3.) Make a copy on your config before editing - Read the release node FOUR times and make sure that you have satisfied every "breaking change" accordingly.

![image](https://user-images.githubusercontent.com/26795429/104941517-76ad0280-59b3-11eb-92af-4073b811fde2.png)

**Comment by Trashmee** [comment]
Wow, there is a lot to play with - worked out of the box. 
 - enable / disable detection, config cleanup, per camera rtmp ffmpeg settings, ...
 - GREAT UI with event history & cool playback UI
 - Like the ability to display & change the mask inline

Thanks a ton to Blake & Paul!

/Edit/
Two observations:
A) Config parameter to default "h=xy" on the UI, the default resolution doesn't look brilliant in my setup. I'll reread the config, maybe I missed something.
   http://192.168.177.177:5000/cameras/frontyard?h=720    # This looks as desired, but would be nice to define the 720 once somewhere
B) picture-entity to display live picture / stream. I had something like the following, which doesn't seem to work with the UI in RC1. Maybe I should check the stream (https://www.home-assistant.io/integrations/stream/) plugin, but that didn't seem to add stability to my HomeAssistant installation.
```
type: picture-glance
camera_view: live
entities: []
image: 'http://192.168.177.177:5000/cameras/frontyard?h=720&bbox=1'
tap_action:
  action: more-info
title: Eingang Live Tagged
```

**Reply by paularmstrong** [reply]
> A) Config parameter to default "h=xy" on the UI, the default resolution doesn't look brilliant in my setup. I'll reread the config, maybe I missed something. http://192.168.177.177:5000/cameras/frontyard?h=720 # This looks as desired, but would be nice to define the 720 once somewhere

This doesn't work in the web UI, only on the API for the camera. Point your browser to `/api/cameras/frontyard?h=720` for that one.

**Reply by TheSleepySlee** [reply]
@paularmstrong I would rather be able to specify the layout of the cameras on the main webpage if possible. Having presets like in tinycam android app would be great.

**Reply by paularmstrong** [reply]
That's going to be pretty far down the line. Configurable layouts is not a simple thing and will require some sort of either localstorage (downside: per browser/device) or a backend store (downside: applies to all browser/devices)

**Reply by Trashmee** [reply]
>> http://192.168.177.177:5000/cameras/frontyard?h=720 
> This doesn't work in the web UI, only on the API for the camera. Point your browser to `/api/cameras/frontyard?h=720` for that one.

Actually it does work :-) But my point is slightly different:

The overview tab (in my environment) has a highres still image of each camera. Probably based on latest.jpg & ?detect? role settings.
    http://192.168.177.177:5000/   # HighRes picture per camera

When selecting a camera the upcoming picture is very low res, 480x360 or 640x360 in my case. 
   http://192.168.177.177:5000/cameras/frontyard
One can alter that manually - but it would be cool to have the 360 configurable some, e.g. in the config.yaml
  http://192.168.177.177:5000/cameras/frontyard?h=720   #works for me - but the URL has to be changed manually

The details view doesn't show pictures at all on my mobile, while it does on the Laptop.
   http://192.168.177.177:5000/events?camera=frontyard&label=person

Don't take that as critismn or feel hurried - I really like where this is heading. 
Big Thanks for the time invested!

**Reply by paularmstrong** [reply]
> The overview tab (in my environment) has a highres still image of each camera. Probably based on latest.jpg & ?detect? role settings.

Yes, this uses `latest.jpg`

> When selecting a camera the upcoming picture is very low res, 480x360 or 640x360 in my case.

This is showing the live `detect` role mjpeg stream. Changing the height won't change the actual quality, it's just going to scale the image

> The details view doesn't show pictures at all on my mobile, while it does on the Laptop.

Fixed in #549

**Reply by Trashmee** [reply]
> Changing the height won't change the actual quality, it's just going to scale the image

Sorry to be stubborn - but it does change the quality A LOT. My detect role on that camera has 2560x1920. 
http://192.168.177.177:5000/cameras/frontyard    - 640x360 - so this is scaled (maybe to fit some layout?) somewhere
http://192.168.177.177:5000/cameras/frontyard?h=1920      - Full 5MP details

I'll shut up now :-)

**Reply by TheSleepySlee** [reply]
@paularmstrong ok, I don't want to overcomplicate things. The UI on mobile phone is nice, with all cams in single column.
I don't know how you choose the size of the pictures displayed on the main page though.

Also, one nice improvement would be to add this feature:
- From main UI, when I click on camera picture it displays the camera selected (already supported)
- Then, when on this page, "Mask & zone creator" is not selected, and I click on the picture, this brings me back to main screen.

This is how it is done on blueiris, and I find it really usefull.

**Reply by paularmstrong** [reply]
> Then, when on this page, "Mask & zone creator" is not selected, and I click on the picture, this brings me back to main screen.

I'm not following the steps here. What do you mean "Mask & zone creator" is "not selected"? It's a link that takes you to another page.

> Sorry to be stubborn - but it does change the quality A LOT. My detect role on that camera has 2560x1920.

You must have some serious CPU power to run detect on such a large view. For this view, I think it makes sense to keep the quality low. Will need another UI for just viewing the full quality live view without the extra processing for bounding boxes, etc. The entire UI is currently meant for debugging purposes only.

**Reply by Trashmee** [reply]
_You must have some serious CPU power to run detect on such a large view. For this view, I think it makes sense to keep the quality low. Will need another UI for just viewing the full quality live view without the extra processing for bounding boxes, etc. The entire UI is currently meant for debugging purposes only._

Understood and accepted.  Just a small feedback - the initial version of your UI has almost everything I'd be looking for to have remote capability to see current state & historic events. 
My point about resolution was a very minor thing, the 'no pictures on mobile' a little bigger. Very happy to see the full featureset on mobile after upgrading to RC2. 

Nice job!

PS: HA + Frigate are running on a 'cheap' j4105 with Coral USB with <50% CPU utilization along with some other containers - nice coding!

**Comment by flaektrem** [comment]
Are you running this as an HA addon or separate docker container? I want to move from separate Docker to HA addon.

Thinking about the fact that i can't configure "shm-size" when running as an addon. I really need that setting (i think?) cause i run 4  RTMP Reolink cams @ 2560x1920 as detector.

**Comment by unrly** [comment]
First - thank you so much for the RC, I'm absolutely loving the UI and this might be the move that makes me get RTSP enabled on all of my Wyze cameras and not just my test camera. My config is extremely simple, using cpu and no args.

The issue I'm having is this:
![image](https://user-images.githubusercontent.com/24898502/105197841-46bc4700-5afa-11eb-9824-a0d173c0f395.png)

Nothing is being loaded on this page and when I go to the direct link, it's just a loading black page: https://<<nabu.casa or internal ip>>/api/hassio_ingress/uTiDTdyLuDijdO3-wJCGyuru-bjA2zmfXhTz4BmncC4/api/grill?

**Reply by paularmstrong** [reply]
Yep. Home Assistant ingress breaks the mjpeg feed: https://github.com/blakeblackshear/frigate/issues/541#issuecomment-763566368

**Reply by unrly** [reply]
Ah ha, thanks!!

**Comment by jmedlin** [comment]
Fantastic work!  I've had it running the last couple days and its been very stable.  I've officially migrated my NVR from Synology Surveillance Station to Frigate, this is 10X better.

**Comment by tubalainen** [comment]
I tried to comment out the `fps: ` for each camera to let Frigate autodetect the frame rate. That did not work for any of my cameras and the CPU went up to over 9.xx on my poor old Core i5 laptop.

The frame rate´s are fixed from each camera (8 fps) but frigate detected it as 21 according to debug.
Anyone else experiencing this? 

Its an easy fix to just uncomment the `fps: ` part again and the load went back to 3.xx
